marine/src/config/to_marine_config.rs

```diff
@@ -124,15 +125,15 @@ impl MModuleConfigBuilder {
             })
             .collect::<HashMap<_, _>>();
 
-        self.config.wasi_envs.extend(mapped_dirs);
+        self.config.wasi_parameters.envs.extend(mapped_dirs);
 
         Ok(self)
     }
 
     fn populate_host_imports(
         mut self,
-        host_imports: HashMap<String, HostImportDescriptor>,
-        call_parameters: Rc<RefCell<CallParameters>>,
+        host_imports: HashMap<String, HostImportDescriptor<WB>>,
+        call_parameters: Arc<Mutex<CallParameters>>, // TODO show mike
```

Comment:
what about mutexes from parking lot?

Reply:
I used the simple Mutex from parking_lot and added comments to all .lock places. `ReentrantMutex` requires `RefCell` inside and it will be too overloaded type.

Reply:
Wasmtime folks just answered that it wasmtime does not spawn threads in runtime, so it should be safe to just do unsafe impl